### ISSUE TITLE
error on api level below 23

### ISSUE THREAD

**Issue by bambolz** [aurelhubert/ahbottomnavigation#110]
when i changed to latest version 1.3.1
it's return error in 
AHBottomNavigation.java  line 131
`return VectorDrawableCompat.create(context.getResources(), drawableRes, null);`
 with error message
parser error

> org.xmlpull.v1.XmlPullParserException: Binary XML file line #2<vector> tag requires viewportWidth > 0

but it works on marshmallow

**Comment by brittonjg** [comment]
Can you give us a bit more info? Are you using vectors? What version of appCompat are you using?

**Comment by aurelhubert** [comment]
As @brittonjg said, we need more details to understand where the problems comes from? Can you give us your Android version, your implementation of the bottom bar (what do you use for the icons? Drawables? Vectors?), etc.

**Comment by bambolz** [comment]
hi @aurelhubert sorry for late response, the error appears when i use ahbottom version 1.3.1 to android version below API 23 (Marshmallow). 

- No I didn't use vectors i still used drawables for the icons. 
- appcompat I used are v7:24.0.0

so i rollback using ahbottom version 1.2.3 and it's works well across API level e.g from 16 to 24

**Comment by mennovogel** [comment]
I had the same issue, and this is indeed fixed when rolling back to version 1.2.3. Here is some extra info: The only code I use is: 
`bottomNavigation.addItem(new AHBottomNavigationItem(R.string.stream_title, R.drawable.ic_stream, android.R.color.white));
bottomNavigation.addItem(new AHBottomNavigationItem(R.string.program_title, R.drawable.ic_program, android.R.color.white));
        bottomNavigation.addItem(new AHBottomNavigationItem(R.string.locations_title, R.drawable.ic_locations, android.R.color.white));
        bottomNavigation.addItem(new AHBottomNavigationItem(R.string.friends_title, R.drawable.ic_friends, android.R.color.white));
        bottomNavigation.addItem(new AHBottomNavigationItem(R.string.info_title, R.drawable.ic_info, android.R.color.white));
        bottomNavigation.setOnTabSelectedListener(this);
        bottomNavigation.setDefaultBackgroundColor(ContextCompat.getColor(getApplicationContext(), R.color.primaryColor));
        bottomNavigation.setAccentColor(Color.WHITE);
        bottomNavigation.setInactiveColor(ContextCompat.getColor(getApplicationContext(), R.color.light_white));
        bottomNavigation.setCurrentItem(0);`
and in my xml: 
`<com.aurelhubert.ahbottomnavigation.AHBottomNavigation
        android:id="@+id/bottom_navigation"
        android:layout_width="match_parent"
        android:layout_height="@dimen/bottom_navigation_height"
        android:layout_gravity="bottom"
        />`

My stacktrace also includes:
android.content.res.Resources$NotFoundException: File res/drawable/abc_ic_ab_back_material.xml from drawable resource ID #0x7f020013

and:
Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #17: invalid drawable tag vector

I use: com.android.support:appcompat-v7:23.3.0

I searched on StackOverflow for the issue, but adding
`android { defaultConfig { vectorDrawables.useSupportLibrary = true } }`
to my project's gradle file didn't fix the issue, changing the appcompat version also didn't fix it. 

I hope you have enough info now, good luck!

**Comment by brittonjg** [comment]
@mennovogel Out of interest, do you have the following at the top of your Activity?

`  static {
    AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
  }`

**Comment by mennovogel** [comment]
No, I didn't have that line.

**Comment by anormobile** [comment]
Same issue

**Comment by microacup** [comment]
add static { AppCompatDelegate.setCompatVectorFromResourcesEnabled(true); } and android { defaultConfig { vectorDrawables.useSuppo
[ERROR.txt](https://github.com/aurelhubert/ahbottomnavigation/files/430280/ERROR.txt)
rtLibrary = true } }  ,But still has error

Reproduce ；chang this line

			`AHBottomNavigationItem item1 = new AHBottomNavigationItem(R.string.tab_1, R.drawable.ic_apps_black_24dp, R.color.color_tab_1);`

 to:

			`AHBottomNavigationItem item1 = new AHBottomNavigationItem(R.string.tab_1, R.drawable.ic_content_add, R.color.color_tab_1);`

and run with adt 22, you can see it.

**Comment by microacup** [comment]
[ERROR.txt](https://github.com/aurelhubert/ahbottomnavigation/files/430288/ERROR.txt)

**Comment by bearprada** [comment]
I thinks the crash is from this commit : https://github.com/aurelhubert/ahbottomnavigation/commit/1b1e321a8748657e7a510588ef15581744fd351e

it's ok to support the VectorDrawable by this way, but it try to catch the wrong exception in this case.

```
    if (drawableRes != 0) {
      try {
        return VectorDrawableCompat.create(context.getResources(), drawableRes, null);
      }catch (Resources.NotFoundException e){
        return ContextCompat.getDrawable(context, drawableRes);
      }
    }
```

please use the `catch(Throwable e)` instead of `catch(Resources.NotFoundException e)`

**Comment by brittonjg** [comment]
@bearprada The reason for that try/catch is to catch the specific error that the resource isn't a VectorDrawable. Therefore we want to use a drawable.

Looking at the error from @microacup it looks more like the VectorDrawable is incorrect. The first error is "Binary XML file line #2<vector> tag requires viewportWidth > 0". I don't think it should be catching invalid SVGs (if it is invalid) as that is a separate issue. Could you drop a link to the code of the VectorDrawable code? How did you create the VectorDrawable?

**Comment by bearprada** [comment]
@brittonjg 
the problem is our SVG file includes `viewportWidth` and `viewportHeight` which is large than `0`. but the `VectorDrawableCompat.create()` is still throw out the exception. let me take look the implementation in `VectorDrawableCompat`

**Comment by brittonjg** [comment]
@bearprada It's strange that it's throwing an exception then. I am slightly hesitant at changing it to a generic error as that doesn't really solve the issue.

**Comment by bearprada** [comment]
@brittonjg I think the problem comes from `VectorDrawableCompat`implementation issue. that means you will get crash when using vector image on Android 4.x devices if you are deal with the vector drawable properly http://stackoverflow.com/questions/35622438/update-android-support-library-to-23-2-0-cause-error-xmlpullparserexception-bin

**Comment by bearprada** [comment]
@brittonjg I found the root cause. it's because we use `<selector>` as the resource file to wrap the vector drawable. so I get the error when parsing resource file by `VectorDrawableCompat`.

**Comment by brittonjg** [comment]
@bearprada That'll be it! What's the solution going forward? Are you able to elaborate on why you're wrapping it within a selector?

**Comment by bearprada** [comment]
just use the SVG drawable without wrapping it by `selector`. e.g,

`selector_ic_activity.xml`
```
<selector xmlns:android="http://schemas.android.com/apk/res/android">
    <item android:drawable="@drawable/ic_activity_s" android:state_checked="true"/>
    <item android:drawable="@drawable/ic_activity"/>
</selector>
```

`ic_activity.xml`
```
<?xml version="1.0" encoding="utf-8"?>
<vector xmlns:android="http://schemas.android.com/apk/res/android"
        android:viewportWidth="48"
        android:viewportHeight="48"
        android:width="48dp"
        android:height="48dp">
    <group
        android:translateX="-40"
        android:translateY="-40">
        <group
            android:translateX="40"
            android:translateY="40">
            <path
                android:pathData="..."
                android:fillColor="@android:color/white" />
        </group>
    </group>
</vector>
```

PREV:
`mNavigationBar.addItem(new AHBottomNavigationItem("Activity", R.drawable.selector_ic_activity));`
NEXT:
`mNavigationBar.addItem(new AHBottomNavigationItem("Activity", R.drawable.ic_activity));`

cc @brittonjg

**Comment by aurelhubert** [comment]
It will be fixed in the next release `2.1.0` by replacing the method `VectorDrawableCompat.create(context.getResources(), drawableRes, null);` with `AppCompatResources.getDrawable(context, drawableRes);`

If you are targeting API below 21, you will have to add this code:
```java
if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
	AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
}
```
